packages/scalar.aspnetcore/Scalar.AspNetCore.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Scalar.AspNetCore
+{
+    public static class OpenApiEndpointRouteBuilderExtensions
+    {
+        public static IEndpointConventionBuilder MapScalarApiReference(this IEndpointRouteBuilder endpoints)
+        {
+            return endpoints.MapGet("/scalar/{documentName}", (string documentName) => Results.Content($$"""
+            <!doctype html>
+            <html>
+            <head>
+                <title>Scalar API Reference -- {{documentName}}</title>
+                <meta charset="utf-8" />
+                <meta name="viewport" content="width=device-width, initial-scale=1" />
+            </head>
+            <body>
+                <script id="api-reference" data-url="/openapi/{{documentName}}.json"></script>

```

Comment:
We provide an option for users to customize the route their OpenAPI document will be served at.

```
app.MapOpenApi(); // use default route of `/openapi/{{documentName}}.json`
app.MapOpenApi("/{{documentName}}/openapi.json");
```

We'll want to consume the data URL in the config object. I assume that will come later.